cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/ObjectMappedTableDefinition.java

```diff
@@ -51,17 +53,23 @@ public ObjectMappedTableDefinition(String name, DatasetDefinition<? extends Tabl
   @Override
   public DatasetSpecification configure(String instanceName, DatasetProperties properties) {
     Map<String, String> props = properties.getProperties();
-    Preconditions.checkArgument(props.containsKey(ObjectMappedTableProperties.TYPE));
+    Preconditions.checkArgument(props.containsKey(ObjectMappedTableProperties.OBJECT_TYPE));
     // schema can normally be derived from the type. However, we cannot use the Type in this method because
     // this is called by the system, where the Type is often not available. for example, if somebody creates
     // an ObjectMappedTable<Purchase> where Purchase is a class internal to their app.
     // we require schema here because we want to validate it to make sure it is supported.
-    Preconditions.checkArgument(props.containsKey(ObjectMappedTableProperties.SCHEMA));
+    Preconditions.checkArgument(props.containsKey(ObjectMappedTableProperties.OBJECT_SCHEMA));
+    Preconditions.checkArgument(props.containsKey(ObjectMappedTableProperties.EXPLORE_KEY_NAME));
+    Preconditions.checkArgument(props.containsKey(ObjectMappedTableProperties.EXPLORE_KEY_TYPE));
     try {
-      Schema schema = ObjectMappedTableProperties.getSchema(props);
-      validateSchema(schema);
+      Schema objectSchema = ObjectMappedTableProperties.getObjectSchema(props);
+      validateSchema(objectSchema);
+      String keyName = ObjectMappedTableProperties.getExploreKeyName(props);
+      Schema.Type keyType = ObjectMappedTableProperties.getExploreKeyType(props);
+      Schema fullSchema = addKeyToSchema(objectSchema, keyName, keyType);
```

Comment:
we'll throw an exception, but it's not a very user friendly one.  I'll just check for it explicitly and throw a more descriptive one.